src/ClockPanel.tsx

```diff
@@ -1,119 +1,84 @@
-      this.initTimers();
-    }
-
-    if (prevData !== data) {
-      this.onPanelRefresh();
-    }
+function getMoment(tz?: string): Moment {
+  if (!tz) {
+    tz = (moment as any).tz.guess();
+  } else {
+    tz = getTemplateSrv().replace(tz);
   }
+  return (moment() as any).tz(tz);
+}
 
-  initTimers = () => {
-    const { refresh } = this.props.options;
-
-    if (this.timerID) {
-      clearInterval(this.timerID);
-      this.timerID = 0;
-    }
-
-    if (refresh === ClockRefresh.dashboard) {
-      return this.tick();
-    }
-
-    const delay = 1000; // 1sec
-    this.timerID = setInterval(
-      () => this.tick(),
-      delay // 1 second or 1 min
-    );
-  };
-
-  onPanelRefresh = () => {
-    const { refresh } = this.props.options;
-    if (refresh === ClockRefresh.dashboard) {
-      this.tick();
-    }
-  };
-
-  componentWillUnmount() {
-    if (this.timerID) {
-      clearInterval(this.timerID);
-      this.timerID = 0;
-    }
+function getTimeFormat(clockType: ClockType, timeSettings: TimeSettings): string {
+  if (clockType === ClockType.Custom && timeSettings.customFormat) {
+    return timeSettings.customFormat;
   }
 
-  tick() {
-    const { timezone } = this.props.options;
-    this.setState({ now: this.getTZ(timezone) });
+  if (clockType === ClockType.H12) {
+    return 'h:mm:ss A';
   }
 
-  getTimeFormat() {
-    const { clockType, timeSettings } = this.props.options;
-
-    if (clockType === ClockType.Custom && timeSettings.customFormat) {
-      return timeSettings.customFormat;
-    }
+  return 'HH:mm:ss';
+}
 
-    if (clockType === ClockType.H12) {
-      return 'h:mm:ss A';
+export function ClockPanel(props: Props) {
+  const [now, setNow] = useState<Moment>(getMoment(props.options.timezone));
+  const { options, width, height } = props;
+  console.log(options);
+  const theme = useTheme2();
+  const { timezone, dateSettings, timezoneSettings, bgColor } = options;
+
+  const className = css`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-flow: column wrap;
+    text-align: center;
+    background-color: ${!bgColor ? theme.colors.background.primary : theme.v1.visualization.getColorByName(bgColor)};
+  `;
+
+  // Clock refresh only on dashboard refresh
+  useEffect(() => {
+    console.log('useEffect');
+    if (props.options.refresh === ClockRefresh.dashboard) {
+      setNow(getMoment(props.options.timezone));
     }
+  }, [props]);
 
-    return 'HH:mm:ss';
-  }
-
-  // Return a new moment instance in the selected timezone
-  // eslint-disable-next-line
-  getTZ(tz?: string): Moment {
-    if (!tz) {
-      tz = (moment as any).tz.guess();
-    } else {
-      tz = getTemplateSrv().replace(tz);
+  // Clock refresh every second
+  useEffect(() => {
+    if (props.options.refresh === ClockRefresh.sec) {
+      const timer = setInterval(() => setNow(getMoment(timezone)), 1000);
+      return () => clearInterval(timer);
     }
```

Comment:
nit: I would probably have done an early exit here instead of having the logic within the if-statement. But ignore this one if you don't feel it would improve readability.

Reply:
it is a matter of style :)